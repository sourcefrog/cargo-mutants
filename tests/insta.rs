// Copyright 2021-2024 Martin Pool

//! Test special handling of cargo-insta.

mod util;
use util::{copy_of_testdata, run};

/// `INSTA_UPDATE=always` in the environment will cause Insta to update
/// the snapshots, so the tests will pass, so mutants will not be caught.
/// This test checks that cargo-mutants sets the environment variable
/// so that mutants are caught properly.
#[test]
fn insta_test_failures_are_detected() {
    for insta_update in ["auto", "always"] {
        println!("INSTA_UPDATE={insta_update}");
        let tmp_src_dir = copy_of_testdata("insta");
        run()
            .arg("mutants")
            .args(["--no-times", "--no-shuffle", "--caught", "-Ltrace"])
            .env("INSTA_UPDATE", insta_update)
            .current_dir(tmp_src_dir.path())
            .assert()
            .success();
    }
}
