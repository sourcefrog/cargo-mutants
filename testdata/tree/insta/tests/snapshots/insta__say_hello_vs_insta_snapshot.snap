---
source: tests/insta.rs
expression: say_hello(name)
---
Hello, Robin!
